doctr/utils/geometry.py

```diff
@@ -201,15 +243,16 @@ def rotate_image(
     # Compute the expanded padding
     if expand:
         exp_shape = compute_expanded_shape(image.shape[:-1], angle)
-        h_pad, w_pad = int(math.ceil(exp_shape[0] - image.shape[0])), int(math.ceil(exp_shape[1] - image.shape[1]))
+        h_pad, w_pad = int(max(0, ceil(exp_shape[0] - image.shape[0]))), int(
+            max(0, ceil(exp_shape[1] - image.shape[1])))
         exp_img = np.pad(image, ((h_pad // 2, h_pad - h_pad // 2), (w_pad // 2, w_pad - w_pad // 2), (0, 0)))
     else:
         exp_img = image
 
     height, width = exp_img.shape[:2]
     rot_mat = cv2.getRotationMatrix2D((width / 2, height / 2), angle, 1.0)
-    rot_img = cv2.warpAffine(exp_img.astype(np.float32), rot_mat, (width, height))
-    if expand:
+    rot_img = cv2.warpAffine(exp_img, rot_mat, (width, height))
+    if preserve_origin_shape:
```

Comment:
I think this needs to be reverted

`if preserve_origin_shape:` should be the conditional check for the rescale inside this "if", or am I missing something?

Reply:
Not sure about what you mean : `if preserve_origin_shape:` is indeed the conditional check for the rescale inside this `if`. Or do you prefer it to be changed to `if expand and preserve_origin_shape:` ?

Reply:
The second option yes

But this means that this line should be kept as is, and only to create an additional `if preserve_origin_shape` inside this loop :+1:

Reply:
ok great !